src/typeGuards.ts

```diff
@@ -2,11 +2,9 @@ import Ajv from "ajv";
 
 import * as schemas from "./_generated/schemas";
 
-const _cachedAjv = new Ajv();
+const _cachedAjv = new Ajv({ allErrors: true });
 
-export type SchemaResult =
-  | { ok: false; errors: Array<Ajv.ErrorObject> }
-  | { ok: true; errors: null };
+export type SchemaResult = { ok: false; errors: Array<Ajv.ErrorObject> } | { ok: true; errors: [] };
```

Comment:
Oh interesting! So with `allErrors:true` it now returns an empty errors array instead of null. I did not know that.

Reply:
That's not it, actually. I changed the type here because it turned out to be very awkward to work with the result when the errors could vary. Typescript couldn't quite do enough in the situations I was working in. Ajv behaves the same.